globus_cli/helpers/local_server.py

```diff
@@ -40,38 +90,49 @@ def is_remote_session():
 
 class RedirectHandler(BaseHTTPRequestHandler):
 
-        def do_GET(self):
+        def do_GET(self):  # noqa
             self.send_response(200)
             self.send_header('Content-type', 'text/html')
             self.end_headers()
-            self.wfile.write(b'You\'re all set, you can close this window!')
 
-            code = parse_qs(urlparse(self.path).query).get('code', [''])[0]
-            self.server.return_code(code)
+            query_params = dict(parse_qsl(urlparse(self.path).query))
+            code = query_params.get('code')
+            if code:
+                self.wfile.write(
+                    bytes(
+                        HTML_TEMPLATE.substitute(post_login_message=DOC_URL),
+                        'ascii'))
+                self.server.return_code(code)
+            else:
+                msg = query_params.get(
+                    'error_description', query_params.get('error'))
+
+                self.wfile.write(
+                    bytes(
+                        HTML_TEMPLATE.substitute(post_login_message=msg),
+                        'ascii'))
```

Comment:
On Python2, this blows up with `TypeError: str() takes at most 1 argument (2 given)`

Because 2-to-3 and encoding and reasons.

I think you can rely on `six.b` for this, but I'm not 100% sure. Checking that now.